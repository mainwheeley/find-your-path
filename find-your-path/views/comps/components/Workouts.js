import React from 'react';
import {
    StyleSheet,
    Text,
    View,
    FlatList,
    Dimensions,
    Button,
} from 'react-native';
import {listWorkouts} from './data.js';
import RecentWorkouts from './recentWorkouts';
import PastWorkouts from './pastWorkouts';

export default class Workouts extends React.Component {

    constructor() {
	super();
    	this.state = {
	    text: 'Useless Placeholder',
	    viewRecent: true,
	    viewPast: false
	};
    }

    renderRecent() {
	if (this.state.viewRecent) {
	    return (
		<RecentWorkouts />
	    )
	}
    }

    renderPast() {
	if (this.state.viewPast) {
	    return (
		<PastWorkouts />
	    )
	}
    }

    setRecentRender=()=> {
	this.setState({
	    viewRecent: true,
	    viewPast: false
	})
    }

    setPastRender=()=> {
	var recent=this.state.viewRecent
	var past=this.state.viewPast
	this.setState({
	    viewRecent: !recent,
	    viewPast: !past
	})
    }


workoutSeparator = () => {
    return (
	<View
	    style={{
		height: 1,
		width: Dimensions.get('window').width,
		backgroundColor: "#000",
	    }}
	/>
    );
}

render() {
return (
  <View style={styles.container}>
    <View style={styles.mainContainer}>
	{this.renderRecent()}
    	{this.renderPast()}
    </View>
    <View style={styles.statsRow}>
        <View style={styles.delButton}>
    	<Button
       	    onPress={this.setPastRender}
       	    title="Switch Recent/Past"
       	    color="#8BC34A"
       	/>
       </View>
    </View>
  </View>);
}

}

const styles = StyleSheet.create({
    delButton: {
	flex: 1,
	height: (Dimensions.get('window').width / 5) - 25,
	alignItems: 'center',
	justifyContent: 'center',
    },
    statsRow: {
	borderColor: '#fff',
	borderTopWidth: 0,
	borderBottomWidth: 0,
	height: Dimensions.get('window').height / 10,
	backgroundColor: '#fff',
	flexDirection: 'row',
	alignItems: 'center',
    },
    itemTitle: {
	padding: 10,
	fontSize: 18,
	height: 44,
	borderBottomColor: '#000',
	fontWeight: 'bold',
    },
    item: {
	padding: 10,
	fontSize: 18,
	height: 44,
	borderBottomColor: '#000',
    },
    mainContainer: {
	justifyContent: 'center',
	height: (Dimensions.get('window').height / 4),
	flex: 1,
	margin: 10
    },
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
  listContainer: {
      marginTop: 14,
      alignSelf: 'stretch',
  },
});
