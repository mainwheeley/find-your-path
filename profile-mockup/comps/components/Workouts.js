import React from 'react';
import { 
    StyleSheet,
    Text,
    View,
    FlatList,
    Dimensions,
    Button,
} from 'react-native';
import {listWorkouts} from './Data';
import recentWorkouts from './recentWorkouts';
import pastWorkouts from './pastWorkouts';

export default class Workouts extends React.Component {

    constructor() {
	super();
    	this.state = { 
	    text: 'Useless Placeholder',
	    viewRecent: true,
	    viewPast: false
	};
    }

    renderRecent() {
	if (this.state.viewRecent) {
	    return (
		<recentWorkouts />
	    )
	}
    }

    renderPast() {
	if (this.state.viewPast) {
	    return (
		<pastWorkouts />
	    )
	}
    }

    setRecentRender=()=> {
	this.setState({
	    viewRecent: true,
	    viewPast: false
	})
    }

    setPastRender=()=> {
	this.setState({
	    viewRecent: false,
	    viewPast: true
	})
    }


workoutSeparator = () => {
    return (
	<View
	    style={{
		height: 1,
		width: Dimensions.get('window').width,
		backgroundColor: "#000",
	    }}
	/>
    );
}

render() {
return (
  <View style={styles.container}>
    <FlatList
	data={listWorkouts}
	ItemSeparatorComponent={this.workoutSeparator}
	renderItem={({ item }) => 
	    <View>
	    <Text style={styles.item}><Text style={styles.itemTitle}>Date: </Text>{item.date}</Text>
	    <Text style={styles.item}><Text style={styles.itemTitle}>Start time: </Text>{item.time}</Text>
	    </View>
	}
    />

    <View style={styles.statsRow}>
        <View style={styles.delButton}>
    	<Button
       	    onPress={this.setPastRender}
       	    title="Delete Account"
       	    color="red"
       	/>
       </View>
    </View>
  </View>);
}

}

const styles = StyleSheet.create({
    delButton: {
	flex: 1,
	height: (Dimensions.get('window').width / 5) - 25,
	alignItems: 'center',
	justifyContent: 'center',
    },
    statsRow: {
	flex: 1,
	flexDirection: 'column',
	flexWrap: 'wrap',
    },
    itemTitle: {
	padding: 10,
	fontSize: 18,
	height: 44,
	borderBottomColor: '#000',
	fontWeight: 'bold',
    },
    item: {
	padding: 10,
	fontSize: 18,
	height: 44,
	borderBottomColor: '#000',
    },
    mainContainer: {
	justifyContent: 'center',
	flex: 1,
	margin: 10
    },
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  listContainer: {
      marginTop: 14,
      alignSelf: 'stretch',
  },
});
