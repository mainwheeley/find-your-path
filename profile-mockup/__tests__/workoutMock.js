import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import Workouts from '../comps/components/Workouts';
import { shallow } from 'enzyme';
describe('Profile Bar', function() {
    it('renders correctly', () => {
	const tree = shallow(<Workouts />)
	expect(tree).toMatchSnapshot();
    });
    it('length 1', function() {
	const tree = shallow(<Workouts />)
	expect(tree.length).toBe(1);
    });
});
