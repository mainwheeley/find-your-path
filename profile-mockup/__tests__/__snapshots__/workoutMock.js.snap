// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Bar renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "height": 333.5,
          "justifyContent": "center",
          "margin": 10,
        }
      }
    >
      <RecentWorkouts />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomWidth": 0,
          "borderColor": "#fff",
          "borderTopWidth": 0,
          "flexDirection": "row",
          "height": 133.4,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": 125,
            "justifyContent": "center",
          }
        }
      >
        <Button
          color="#8BC34A"
          onPress={[Function]}
          title="Switch Recent/Past"
        />
      </View>
    </View>
  </View>,
  "nodes": Array [
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "height": 333.5,
            "justifyContent": "center",
            "margin": 10,
          }
        }
      >
        <RecentWorkouts />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderBottomWidth": 0,
            "borderColor": "#fff",
            "borderTopWidth": 0,
            "flexDirection": "row",
            "height": 133.4,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "height": 125,
              "justifyContent": "center",
            }
          }
        >
          <Button
            color="#8BC34A"
            onPress={[Function]}
            title="Switch Recent/Past"
          />
        </View>
      </View>
    </View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Workouts />,
    "_instance": Workouts {
      "context": Object {},
      "props": Object {},
      "refs": Object {},
      "setPastRender": [Function],
      "setRecentRender": [Function],
      "state": Object {
        "text": "Useless Placeholder",
        "viewPast": false,
        "viewRecent": true,
      },
      "updater": Updater {
        "_renderer": [Circular],
      },
      "workoutSeparator": [Function],
    },
    "_newState": null,
    "_rendered": <View
      style={
        Object {
          "backgroundColor": "#fff",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "height": 333.5,
            "justifyContent": "center",
            "margin": 10,
          }
        }
      >
        <RecentWorkouts />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderBottomWidth": 0,
            "borderColor": "#fff",
            "borderTopWidth": 0,
            "flexDirection": "row",
            "height": 133.4,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "height": 125,
              "justifyContent": "center",
            }
          }
        >
          <Button
            color="#8BC34A"
            onPress={[Function]}
            title="Switch Recent/Past"
          />
        </View>
      </View>
    </View>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Workouts />,
}
`;
